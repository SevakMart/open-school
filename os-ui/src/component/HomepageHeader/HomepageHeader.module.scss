@import "../../styles/Mixins/ButtonMixin.scss";
@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";
.headerContainer {
  background-color: #eeeff4;
  .mainContent {
    @include flexStyle(row, center, space-between);
    padding: 7% 9.5% 7% 10.7%;
    background-color: #eeeff4;
    .leftContent {
      width: 40%;
      @include flexStyle(column, flex-start, space-between);
      h1 {
        display: inline;
        @include fontStyle("Montserrat", normal, 800, 3rem, 3.688rem);
        color: #5e617b;
        &::before {
          content: "Free ";
          color: #222427;
        }
      }
      p {
        display: inline;
        @include fontStyle("Montserrat", normal, 400, 1rem, 1.5rem);
        color: #5e617b;
      }
      .buttonContainer {
        width: 60%;
        @include flexStyle(row, center, space-between);
        & button:nth-of-type(1) {
          @include buttonStyle(
            #5e617b,
            #ffffff,
            1rem,
            none,
            4rem,
            2rem,
            2rem,
            1.375rem,
            1.375rem
          );
          @include fontStyle("Montserrat", normal, 500, 1rem, 1.25rem);
          text-transform: uppercase;
        }
        & button:nth-of-type(2) {
          @include buttonStyle(
            transparent,
            #5e617b,
            1rem,
            none,
            4rem,
            2rem,
            2rem,
            1.375rem,
            1.375rem
          );
          @include fontStyle("Montserrat", normal, 500, 1rem, 1.25rem);
          text-transform: uppercase;
          border-color: #5e617b;
        }
      }
    }
    .rightContent {
      width: 50%;
      text-align: center;
    }
  }
}
