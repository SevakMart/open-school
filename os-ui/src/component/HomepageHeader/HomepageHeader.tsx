import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import Navbar from '../Navbar/Navbar';
import HomepageMainImage from '../../assets/svg/HomepageMainImage.svg';
import Button from '../Button/Button';
import { Types } from '../../types/types';
import { openModal } from '../../redux/Slices/PortalOpenStatus';
import styles from './HomepageHeader.module.scss';

const HomepageHeader = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const {
    mainContent, leftContent, rightContent, buttonContainer,
  } = styles;

  const handleSignUp = () => {
    dispatch(openModal(Types.Button.SIGN_UP));
  };

  const handleSignIn = () => {
    dispatch(openModal(Types.Button.SIGN_IN));
  };

  return (
    <>
      <Navbar />
      <div className={mainContent}>
        <div className={leftContent}>
          <h1>{t('string.homePage.header.educationalPlatform')}</h1>
          <p>{t('string.homePage.header.chooseCategories')}</p>
          <div className={buttonContainer}>
            <Button.SignUpButton className={['headerButton', 'headerButton__signUp']} onClick={handleSignUp}>
              {t('button.homePage.signUp')}
            </Button.SignUpButton>
            <Button.SignInButton className={['headerButton', 'headerButton__signIn']} onClick={handleSignIn}>
              {t('button.homePage.signIn')}
            </Button.SignInButton>
          </div>
        </div>
        <div className={rightContent}>
          <img src={HomepageMainImage} alt="Education platform logo" />
        </div>
      </div>
    </>
  );
};
export default HomepageHeader;
