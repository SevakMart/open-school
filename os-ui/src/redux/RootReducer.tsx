import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
