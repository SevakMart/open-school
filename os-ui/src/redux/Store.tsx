import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
