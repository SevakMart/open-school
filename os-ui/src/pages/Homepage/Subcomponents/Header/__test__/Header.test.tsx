import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import i18next from 'i18next';
import { initReactI18next, I18nextProvider } from 'react-i18next';
import { Suspense } from 'react';
import HomepageHeader from '../Header';
import { store } from '../../../../../redux/Store';

i18next.use(initReactI18next).init({
  lng: 'en',
  fallbackLng: 'en',
  interpolation: {
	  escapeValue: false,
  },
});

describe('Create tests for HomepageHeader', () => {
  test('Create snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18next}>
          <Suspense fallback={<div>Loading...</div>}>
            <HomepageHeader />
          </Suspense>
        </I18nextProvider>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
