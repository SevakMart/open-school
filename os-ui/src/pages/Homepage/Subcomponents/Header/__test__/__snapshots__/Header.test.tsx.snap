// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create tests for HomepageHeader Create snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="navContainer"
  >
    <h2>
      string.homePage.navBar.logo
    </h2>
    <div
      class="categoriesMentors"
    >
      <p>
        string.homePage.navBar.exploreCategories
      </p>
      <p>
        string.homePage.navBar.mentors
      </p>
    </div>
    <div
      class="buttonContent"
    >
      <p>
        button.homePage.becomeMentor
      </p>
      <button
        class="navbarSignInButton"
        type="button"
      >
        button.homePage.signIn
      </button>
    </div>
  </nav>
  <div
    class="mainContent"
  >
    <div
      class="leftContent"
    >
      <h1>
        string.homePage.header.educationalPlatform
      </h1>
      <p>
        string.homePage.header.chooseCategories
      </p>
      <div
        class="buttonContainer"
      >
        <button
          class="headerButton headerButton__signUp"
          type="button"
        >
          button.homePage.signUp
        </button>
        <button
          class="headerButton headerButton__signIn"
          type="button"
        >
          button.homePage.signIn
        </button>
      </div>
    </div>
    <div
      class="rightContent"
    >
      <img
        alt="Education platform logo"
        src="HomepageMainImage.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
